### PR TITLE
Android Node Query Fixes

### PR DESCRIPTION
**Pull request by asutula**
Android got a little out of whack with the last node update. This just fixed the problems.

### CODE REVIEW

File: android/app/src/main/java/com/textile/textilenode/TextileNode.java [hunk 1]
```diff
@@ -241,12 +246,11 @@ public void getPhotoBlocks (String offset, Integer limit, String threadName, Pro
...
-            JSONArray jsonArray = obj.getJSONArray("items");
-            for (int i = 0; i < jsonArray.length(); i++) {
-                Object value = jsonArray.get(i);
-                array.pushString((String) value);
+            JSONArray jsonArray = obj.optJSONArray("items");
+            if (jsonArray == null) {
+                jsonArray = new JSONArray();
             }
+            WritableArray array = TextileNode.convertJsonToArray(jsonArray);
             promise.resolve(array);
```
Comment (asutula) [inline android/app/src/main/java/com/textile/textilenode/TextileNode.java]: Previously, we were sending an array of `String`s here. Needed to make it consistent with the iOS version which sends an array of objects that will be properly converted to JS objects by RN. In this case, that object type is `WritableArray` (which is an array of `WritableMap`s). RN automatically converts them to JS objects.

File: android/app/src/main/java/com/textile/textilenode/TextileNode.java [hunk 2]
```diff
@@ -289,7 +293,8 @@ public void pairDevice (String pkb64, Promise promise) {
     public void getFilePath(String uriString, Promise promise) {
         Uri uri = Uri.parse(uriString);
         try {
-            promise.resolve(RealPathUtil.getRealPath(reactContext, uri));
+            String path = RealPathUtil.getRealPath(reactContext, uri);
+            promise.resolve(path);
```
Comment (asutula) [inline android/app/src/main/java/com/textile/textilenode/TextileNode.java]: Noticed some weirdness here. Seems to work fine on a real device, but on the emulator, the path returned here eventually is identified as 'file doesn't exist'. Anyone seen that before?

Comment (asutula) [reply android/app/src/main/java/com/textile/textilenode/TextileNode.java]: @andrewxhill ^?

File: android/app/src/main/java/com/textile/textilenode/TextileNode.java [hunk 3]
```diff
@@ -300,4 +305,109 @@ private static void emitDeviceEvent(String eventName, @Nullable WritableMap even
         // https://facebook.github.io/react-native/docs/native-modules-android.html#sending-events-to-javascript
         reactContext.getJSModule(DeviceEventManagerModule.RCTDeviceEventEmitter.class).emit(eventName, eventData);
     }
+
```
Comment (asutula) [inline android/app/src/main/java/com/textile/textilenode/TextileNode.java]: These are some utility methods to convert to/from `JSONObject` and `WritableMap/Array`. Probably a better place for them?

File: android/app/src/main/java/com/textile/textilenode/TextileNode.java (hunk 2 above)
Comment (andrewxhill) [reply android/app/src/main/java/com/textile/textilenode/TextileNode.java]: haven't seen it, no... android only?

Comment (asutula) [reply android/app/src/main/java/com/textile/textilenode/TextileNode.java]: Yea android only.

**Comment by asutula** [comment]
Closing this because it's included in #165

**Pull request closed without merging**